File: src/main/java/kanti/sl/arguments/MutableStateArgumentsImpl.java
package kanti.sl.arguments;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.*;

public class MutableStateArgumentsImpl implements MutableStateArguments {

	private final Map<String, MutableStateArgument> mapArgs;

	public MutableStateArgumentsImpl(@NotNull MutableStateArgument... args) {
		mapArgs = new HashMap<>();
		for (MutableStateArgument arg : args) {
			mapArgs.put(arg.getKey(), arg);
		}
	}

	public MutableStateArgumentsImpl() {
		mapArgs = new HashMap<>();
	}

	@NotNull
	@Override
	public List<MutableStateArgument> getArguments() {
		return new ArrayList<>(mapArgs.values());
	}

	@Nullable
	@Override
	public MutableStateArgument get(@NotNull String key) {
		return mapArgs.get(key);
	}

	@NotNull
	@Override
	public MutableStateArguments put(@NotNull MutableStateArgument arg) {
		mapArgs.put(arg.getKey(), arg);
		return this;
	}

}

File: src/main/kotlin/kanti/sl/arguments/MutableStateArgumentsImpl.kt
package kanti.sl.arguments

class MutableStateArgumentsImpl(
	arguments: Iterable<StateArgument> = listOf()
) : MutableStateArguments {

	private val argMap: MutableMap<String, StateArgument> = HashMap()

	init {
		for (arg in arguments) {
			argMap.put(arg.name, arg)
		}
	}

	override fun iterator(): Iterator<StateArgument> {
		return argMap.values.iterator()
	}

	override fun put(argument: StateArgument) {
		argMap[argument.name] = argument
	}

	override operator fun get(key: String): StateArgument? {
		return argMap[key]
	}
}